File: kh1_src/tabs/journal.py
from dash import Dash, html, dcc, callback, Input, Output, State, ALL
import kh1_src.kh1_utils as utils

from kh1_src.tabs.journal_tabs import *


@callback(
    Output("JournalDiv", "children"),
    Input("JournalTabs", "value"),
)
def __create_journal(tab):
    if tab == "Chronicles":
        return create_chronicles()
    if tab == "Ansem's Report":
        return create_reports()
    if tab == "Characters":
        return create_journal_characters()
    if tab == "101 Dalmatians":
        return create_dalmatians()
    if tab == "Trinity List":
        return create_trinity()

def create_journal():
    jtabs = dcc.Tabs(id="JournalTabs", value="Journal Flags")
    jtabs.children = [
        dcc.Tab(label="Journal Flags", value="Journal Flags"),
        dcc.Tab(label="Chronicles", value="Chronicles"),
        dcc.Tab(label="Ansem's Report", value="Ansem's Report"),
        dcc.Tab(label="Characters", value="Characters"),
        dcc.Tab(label="101 Dalmatians", value="101 Dalmatians"),
        dcc.Tab(label="Trinity List", value="Trinity List"),
        dcc.Tab(label="Mini Games", value="Mini Games"),
        dcc.Tab(label="Treasures", value="Treasures"),
    ]
    return html.Div([
        jtabs,
        html.Div(id="JournalDiv"),
    ])

File: kh1_src/tabs/journal.js
import * as tabs from "./journal_tabs/index.js"

export function create_journal() {
    const kh1div = document.getElementById("kh1div");
    kh1div.innerHTML = `
    <div>
        <div>
            <h3>Entry:</h3>
            <select id="kh1jtabs">
                <option value="Journal Flags">Journal Flags</option>
                <option value="Chronicles">Chronicles</option>
                <option value="Ansem's Report">Ansem's Report</option>
                <option value="Characters">Characters</option>
                <option value="101 Dalmatians">101 Dalmatians</option>
                <option value="Trinity List">Trinity List</option>
                <option value="Mini Games">Mini Games</option>
                <option value="—Battle Record—">—Battle Record—</option>
                <option value="Treasures">Treasures</option>
                <option value="Synthesis">Synthesis</option>
            </select>
        </div>
        <div id="kh1jdiv"></div>
    </div>`;
    const kh1jtabs = document.getElementById("kh1jtabs");
    const kh1jdiv = document.getElementById("kh1jdiv");
    function tab_sel() {
        switch (kh1jtabs.value) {
            case "Journal Flags": {
                tabs.create_flags();
                break;
            }
            case "Chronicles": {
                tabs.create_chronicles();
                break;
            }
            case "Ansem's Report": {
                tabs.create_reports();
                break;
            }
            case "Characters": {
                tabs.create_journal_characters();
                break;
            }
            case "101 Dalmatians": {
                tabs.create_dalmatians();
                break;
            }
            case "Synthesis": {
                tabs.create_synthesis();
                break;
            }
            default: {
                kh1jdiv.innerHTML = "";
                break;
            }
        }
    }
    kh1jtabs.addEventListener("change", tab_sel);
    tab_sel();
}
